### PR TITLE
68 purge angular 10

### PR DESCRIPTION
**Pull request by newtewt**
# Description

async/await and angular 10 updates for purge test

https://github.com/medic/angular10-migration/issues/68

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: tests/page-objects/reports/reports.po.js [hunk 1]
```diff
@@ -18,7 +18,8 @@ module.exports = {
   firstReport: () => element(by.css(`${reportListID} li:first-child`)),
   listLoader: () => element(by.css(`${reportListID} .loader`)),
   list: () => element(by.css(reportListID)),
-  reportByUUID: uuid => module.exports.list().all(by.css('li[data-record-id="' + uuid + '"]')),
+  reportByUUID: uuid => module.exports.list().all(by.css('li[data-record-id="' + uuid + '"]')).first(),
+  reportsByUUID: uuid => module.exports.list().all(by.css('li[data-record-id="' + uuid + '"]')),
```
Comment (craig-landry) [inline tests/page-objects/reports/reports.po.js:22]: Some template string goodness to tidy that up and make sure we don't end up in quoting trouble, and then extracting to a variable to be sure we're conveying that they're the same thing:

```suggestion
  const reportUUIDSelector = `li[data-record-id="${uuid}"]`;
  reportByUUID: uuid => module.exports.list().all(by.css(reportUUIDSelector)).first(),
  reportsByUUID: uuid => module.exports.list().all(by.css(reportUUIDSelector)),
```

**Review by craig-landry — CHANGES_REQUESTED** [review]

File: tests/page-objects/common/common.po.js [hunk 2]
```diff
@@ -195,6 +208,12 @@ module.exports = {
     helper.waitUntilReady(hamburgerMenuOptions);
```
[the remark this answers is not in the document]
Comment (newtewt) [reply tests/page-objects/common/common.po.js:208]: I added the return here. However, I was a bit reluctant as I didn't use this code. I'll ensure the tests pass before merging.

File: tests/page-objects/reports/reports.po.js [hunk 3]
```diff
@@ -18,7 +18,8 @@ module.exports = {
   firstReport: () => element(by.css(`${reportListID} li:first-child`)),
   listLoader: () => element(by.css(`${reportListID} .loader`)),
   list: () => element(by.css(reportListID)),
-  reportByUUID: uuid => module.exports.list().all(by.css('li[data-record-id="' + uuid + '"]')),
+  reportByUUID: uuid => module.exports.list().all(by.css(`li[data-record-id="${uuid}"]`)).first(),
+  reportsByUUID: uuid => module.exports.list().all(by.css(`li[data-record-id="${uuid}"]`)),
```
Comment (craig-landry) [inline tests/page-objects/reports/reports.po.js:22]: I see you got the string template stuff, but perhaps forgot about the request to extract to a variable. I was going to suggest this...
```
  const matchedListItems = by.css(`li[data-record-id="${uuid}"]`);
  reportByUUID: uuid => module.exports.list().all(matchedListItems).first(),
  reportsByUUID: uuid => module.exports.list().all(matchedListItems),
```
...but then realized this is all inline. I'm tempted to say "eh, nevermind, let's leave it for now" but this is still pretty quick to make easier on the reader. Similar to how lines 15 and 16 work, please just make functions for `reportByUUID` and reportsByUUID with the logic in them (not all inline here). Then, a nice benefit is that the implementation of `reportByUUID` will just be to call `reportsByUUID` and then call `first` on the result.

**Review by craig-landry — CHANGES_REQUESTED** [review]

**Comment by newtewt** [comment]
I updated to pull the function out and just make calls to that instead of in-lining. Last review maybe @craig-landry ?

**Review by craig-landry — APPROVED** [review]
lgtm. thanks for doing the cleanup work on existing stuff along the way!

**Pull request merged**